guardian/transcription-service
# Add media download service (part 1)

## What does this change?
This pr introduces a new app to the transcription service - 'media download'. This service reads off a (new) SQS queue, parses the messages into a new `MediaDownloadJob` type, and then downloads the relevant url, then uploads the resulting file to the 'source media bucket' which is used by the transcription service, and sends an SQS message to the transcription worker to tell it to transcribe the file

To facilitate testing this service, I've also added a new endpoint to the transcription API, which pushes messages onto the queue based off the request body. You can test this endpoint by making a POST request to localhost:9103/api/transcribe-url with the following body:

```
{"url":"https://www.youtube.com/watch?v=upMOIAwWXmM","languageCode":"en","translationRequested":false}
```

Note - the infrastructure for the media download service doesn't exist yet, the project can only be run locally at the moment. 

## How to test
You'll need to manually install yt-dlp (https://github.com/yt-dlp/yt-dlp?tab=readme-ov-file#installation) - probably `pip install yt-dlp` is the easiest if you've got a suitable pythong environment set up. I haven't scripted this step because eventually we'll run yt-dlp using a docker container rather than isntalling it directly onto dev machines. This is how the existing whisper process works

If you run the transcription API, send the above request. If you get a 200 back, then run `npm run media-download::start` and you should see it read sqs/download the video/upload file to s3/send a transcription sqs message

## Still todo
 - UI for providing youtube url
 - Infra for media download service
 - Update output handler and export UI to support saving the original video to google drive
